docs/reference/modules/discovery/no-master-block.asciidoc

```diff
@@ -0,0 +1,22 @@
+[[no-master-block]]
+=== No master block
+
```

Comment:
I think this content should actually be folded into a bigger discovery settings page (e.g. discovery-settings) rather than stand alone.

Reply:
Yes, on reflection I think we should collect the reference definitions of many of the settings into a single place, but I would rather do so as a followup and not in this PR.

Reply:
I created this PR while this was all fresh in my mind: https://github.com/elastic/elasticsearch/pull/36909